src/poetry/config/config.py

```diff
@@ -170,11 +171,17 @@ def process(self, value: Any) -> Any:
         if not isinstance(value, str):
             return value
 
-        return re.sub(
-            r"{(.+?)}",
-            lambda m: self.get(m.group(1)),  # type: ignore[no-any-return]
-            value,
-        )
+        def resolve_from_config(match: re.Match[str]) -> str | Any:
```

Comment:
`str | Any` is pointless, it might as well be `Any`.

(I don't understand this change, this comment is strictly on the type annotation.)

Reply:
You are right, corrected.